libs/api_common/windows_platform_common.cpp

```diff
@@ -420,31 +420,10 @@ get_program_type_windows(const std::string& section, const std::string&)
         }
     }
 
-    // Note: Ideally this function should throw an exception whenever a matching ProgramType
-    // is not found, but that causes a problem in the following scenario:
-    //
-    // This function is called by verifier code in 2 cases:
-    //   1. When verifying the code
-    //   2. When parsing the ELF file and unmarshalling the code.
-    // For the second case mentioned above, if the ELF file contains an unknown section name (".text", for example),
-    // and this function is called while unmarshalling that section, throwing an exception here
-    // will fail the parsing of the ELF file.
-    //
-    // Hence this function returns ProgramType for EBPF_PROGRAM_TYPE_UNSPECIFIED when verification is not
-    // in progress, and throws an exception otherwise.
     try {
-        return get_program_type_windows(*global_program_type);
+        return *get_program_type_windows(*global_program_type);
     } catch (...) {
-        if (!get_verification_in_progress()) {
-            return PTYPE("unspec", {0}, (uint64_t)&EBPF_PROGRAM_TYPE_UNSPECIFIED, {});
-        } else {
-            if (global_program_type_found) {
-                auto guid_string = guid_to_string(global_program_type);
-                throw std::runtime_error(std::string("ProgramType not found for GUID ") + guid_string);
-            } else {
-                throw std::runtime_error(std::string("ProgramType not found for section " + section));
-            }
-        }
+        return PTYPE("unspec", {0}, (uint64_t)&EBPF_PROGRAM_TYPE_UNSPECIFIED, {});
```

Comment:
when verification is in progress, and this function returns `EBPF_PROGRAM_TYPE_UNSPECIFIED`, verifier will try to verify the BPF program using `EBPF_PROGRAM_TYPE_UNSPECIFIED`. Is that desirable?

Reply:
This was mainly the reason this function used to throw exception when verification was in progress to ensure verifier does not try to verify the program using a wrong `EbpfProgramType`

Reply:
Which code path are you referring to?    See line 164 of api_common.cpp where ebpf_verify_program() will not call analyze() at line 168 if the type is unspecified.  So I don't think it's true in this PR that "verifier will try to verify the BPF program..."